experiment/template/main_eVOLVER.py

```diff
@@ -1,92 +1,149 @@
+    logger.debug('elapsed time: %.4f hours' % elapsed_time)
+    print("{0}: {1} Hours".format(custom_script.EXP_NAME ,elapsed_time))
     OD_data, temp_data = eVOLVER_module.read_data(vials)
-    skip = False
     if OD_data is not None and temp_data is not None:
         if OD_data == 'empty':
             print("Data Empty! Skipping data log...")
+            logger.warning('empty OD data received! Skipping data log...')
         else:
             for x in vials:
                 OD_data[x] = OD_data[x] - OD_initial[x]
         eVOLVER_module.parse_data(OD_data, elapsed_time, vials, 'OD')
         eVOLVER_module.parse_data(temp_data, elapsed_time, vials, 'temp')
 
-        #Make decision
+        # make decision
         custom_functions(elapsed_time)
 
-        #Save Variables
-        global run_exp
+        # save Variables
         eVOLVER_module.save_var(start_time, OD_initial)
-    run_exp = root.after(1000,update_eVOLVER)
-
 
-#### Custom Script
 def custom_functions(elapsed_time):
     global OD_data, temp_data
     if OD_data == 'empty':
         print("UDP Empty, did not execute program!")
+        logger.warning('empty OD data received! Skipping custom routine...')
     else:
-        ###load script from another python file
+        # load user script from custom_script.py
         if custom_script.OPERATION_MODE == 'turbidostat':
             custom_script.turbidostat(OD_data, temp_data, vials, elapsed_time)
         if custom_script.OPERATION_MODE == 'chemostat':
             custom_script.chemostat(OD_data, temp_data, vials, elapsed_time)
 
-
-#### Runs if this is main script
 if __name__ == '__main__':
-    vials = range(0,16)
-    start_time, OD_initial = eVOLVER_module.initialize_exp(vials)
-    root=Tk()
-    make_GUI(root)
-    update_eVOLVER()
-    root.mainloop()
+    options = get_options()
+
+    #changes terminal tab title in OSX
+    print('\x1B]0;eVOLVER EXPERIMENT: PRESS Ctrl-C TO PAUSE\x07')
+
+    # silence logging until experiment is initialized
+    logging.basicConfig(level=logging.CRITICAL + 10)
+
+    vials = range(0, 16)
+    start_time, OD_initial = eVOLVER_module.initialize_exp(vials,
+                                                 always_yes=options.always_yes)
+
+    # logging setup
+    if options.quiet:
+        logging.basicConfig(level=logging.CRITICAL + 10)
+    else:
+        if options.verbose == 0:
+            level = logging.INFO
+        elif options.verbose >= 1:
+            level = logging.DEBUG
+        logging.basicConfig(format='%(asctime)s - %(name)s - [%(levelname)s] '
+                            '- %(message)s',
+                            datefmt='%Y-%m-%d %H:%M:%S',
+                            filename=os.path.join(custom_script.EXP_NAME,
+                                                  options.log_name),
+                            level=level)
+
+    loop_start = time.time()
+    while True:
+        try:
+            # run update if at least a 10th of second has passed
```

Comment:
Why does it need to wait 0.1 s?

Reply:
I wasn't sure if it was a good idea to run the loop at high rates, but I can reduce this a 100th of a second...